src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/BlahVetCreation.java

```diff
@@ -0,0 +1,249 @@
+import org.broadinstitute.hellbender.engine.ReadsContext;
+import org.broadinstitute.hellbender.engine.ReferenceContext;
+import org.broadinstitute.hellbender.engine.VariantWalker;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.SortedSet;
+
+/**
+ * Example/toy program that shows how to implement the VariantWalker interface. Prints supplied variants
+ * along with overlapping reads/reference bases/variants (if present).
+ */
+@CommandLineProgramProperties(
+        summary = "Example tool that prints variants supplied to the specified output file (stdout if none provided), along with overlapping reads/reference bases/variants (if provided)",
+        oneLineSummary = "Example tool that prints variants with optional contextual data",
+        programGroup = ExampleProgramGroup.class,
+        omitFromCommandLine = true
+)
+public final class BlahVetCreation {
+
+    /**
+     * Expected headers for the Variant Table (VET)
+     */
+    public enum HeaderFieldEnum {
+        // TODO is this where the validation step (required vs not) lives  -- fail if there is missing data for a required field
+        // and just leave it empty if not required
+
+        START_POSITION("start_position") { // Required
+             public String getColumnValue(final VariantContext variant) throws IOException {
+                 final String startPosition = String.valueOf(variant.getStart());
+                 if (startPosition.equals(null)) {
+                     throw new IllegalArgumentException("Cannot be missing required value for start_position");
+                 }
+                 return startPosition; // TODO see what James' writer does
+            }
+        },
+
+        REFERENCE_BASES("reference_bases") { // Required
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                final String referenceBase = variant.getReference().getBaseString();
+                if (referenceBase.equals(null)) {
+                    throw new IllegalArgumentException("Cannot be missing required value for reference_bases");
+                }
+                return referenceBase;
+            }
+        },
+
+        /*ALTERNATE_BASES("alternate_bases") { // TODO why are these RECORD in BQ? Try with delimited strings
+            public List<Allele> getColumnValue(final VariantContext variant) throws IOException {
+                return variant.getAlternateAlleles();
+            }
+        },*/
+
+        ALTERNATE_BASES_ALT("alternate_bases.alt") {
+            public String getColumnValue(final VariantContext variant) throws IOException {
+                return variant.getAlternateAllele(0).getBaseString();
```

Comment:
anything under the `alternate_bases` record type will return an array of values because there will be a value for each alternate base.  Depending on if we choose to output json/csv/avro we'll have to format that array of values differently.  So say we were using csv and we were respresenting an array of values as a `:` separated string.  This would want to look something like.

```
String output = ""
for(Allele a : variant.getAlternateAlleles()) {
  output += a.getBaseString() + ":"
}
return output
```

youd want to actually make this a function and make it better but you get the gist